third_party/tonic/dart_wrappable.h

```diff
@@ -43,7 +43,11 @@ class DartWrappable {
 
   virtual void ReleaseDartWrappableReference() const = 0;
 
+
+  // Use this method sparingly. It follows a slower path using Dart_New.
+  // Prefer constructing the object in Dart code and using ClaimDartHandle.
   Dart_Handle CreateDartWrapper(DartState* dart_state);
+  void ClaimDartHandle(Dart_Handle wrappable);
```

Comment:
Prefer an RAII wrapper that takes a handle and claims it on collection. It is easy to forget claiming this handle and cause a leak.

Reply:
You'll have to rewrite the `ClaimDartHandle` routine to ignore null wrappers.

Reply:
How would the RAII wrapper work?  `ClaimDartHandle` binds a C++ DartWrappable object to a Dart object.  The binding lasts until the Dart object is garbage collected, at which point the C++ object is dereffed.

`ClaimDartHandle` is similar to the existing `DartWrappable::AssociateWithDartWrapper` method used during construction of Dart objects from C++.  I'd recommend replacing `AssociateWithDartWrapper` with `ClaimDartHandle`.  `DartCallConstructor` would then need to be changed to extract the Dart `this` reference and pass it to `ClaimDartHandle`.

Reply:
Same question. 

The DartWrappable _will_ claim the handle on destruction currently, but we still need to tell it at some point about the Dart_Handle that it needs to track as its Dart peer.

Reply:
Ah, I read "Claim" as "Collect" or "Destroy" instead of "claim as a peer".

Reply:
If there's a better name I'm for it - Associate was already taken, and does something a bit different - and I didn't want to break it in the mean time.

Reply:
Maybe `SetDartPeer`?

Reply:
Or just keep the name `AssociateWithDartWrapper` and change it to take a `Dart_Handle`.  The part of `AssociateWithDartWrapper` that grabs the `Dart_Handle` from the `Dart_NativeArguments` would then be moved into `DartCallConstructor`.

Reply:
Ahhh ok, I see now. That SG.

Reply:
Argh.  I forgot I had the lable on this.  I'll open a new PR with that rename applied.